api/v1alpha1/datadogagent_default.go

```diff
@@ -920,15 +930,23 @@ func DefaultDatadogFeatureNetworkMonitoring(ft *DatadogFeatures) *NetworkMonitor
 func DefaultDatadogAgentSpecClusterAgent(clusterAgent *DatadogAgentSpecClusterAgentSpec) *DatadogAgentSpecClusterAgentSpec {
 	if IsEqualStruct(*clusterAgent, DatadogAgentSpecClusterAgentSpec{}) {
 		clusterAgent.Enabled = NewBoolPointer(true)
-		return clusterAgent
+
+		if !BoolValue(clusterAgent.Enabled) {
```

Comment:
This will never be the case I believe

Reply:
Agree, it was to follow the same pattern everywhere.